core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/commit/CommitInfoSearchRequest.java

```diff
@@ -88,11 +100,100 @@ protected CommitInfos createEmptyResult(int limit) {
 		return new CommitInfos(limit, 0);
 	}
 	
-	private void addBranchClause(final ExpressionBuilder builder) {
+	private void addSecurityFilter(final ExpressionBuilder builder, RepositoryContext context) {
+		
+		final User user = context.service(User.class);
+		if (user.isAdministrator() || user.hasPermission(Permission.requireAll(Permission.OPERATION_BROWSE, Permission.ALL))) {
+			if (containsKey(OptionKey.BRANCH)) {
+				addBranchClause(builder, context, null, null, true);		
```

Comment:
I know I said the opposite earlier, but after seeing it I think it would be better to separate the branch filter and security filter into two separate filters instead. Reason: the number of `if -> return` combinations are just too much within this method.